quarkchain/cluster/root_state.py

```diff
@@ -367,13 +372,25 @@ def validate_block(self, block, block_hash=None):
 
         # Check coinbase
         if not self.env.quark_chain_config.SKIP_ROOT_COINBASE_CHECK:
-            coinbase_amount = self.__calculate_root_block_coinbase(block)
-            if coinbase_amount != block.header.coinbase_amount:
+            expected_coinbase_amount = self.__calculate_root_block_coinbase(block)
+            actual_coinbase_amount = block.header.coinbase_amount_map.balance_map.get(
+                self.env.quark_chain_config.genesis_token, 0)
+
+            if len(block.header.coinbase_amount_map.balance_map) > 1:
+                raise ValueError("Incorrect coinbase_amount_map: too many tokens")
+
+            if (
```

Comment:
in the future we also expect multiple tokens in coinbase balance map, right? if so, can add a TODO comment here for this check to clarify